updates to UI wording and calling out more clearly why usages you know exist might not show in the version controlled usages table

LeeMeyer opened the pull request:
this relates to
https://github.com/OctopusDeploy/ResearchAndDevelopment/issues/174

there have been some bug fixes but also some changes to copy since https://github.com/OctopusDeploy/Issues/issues/7399 it wasn't clear enough why usages they expect to see my not show.

harrisonmeister commented on docs/projects/custom-step-templates.md:
```diff
@@ -56,7 +56,7 @@ You'll also see a handy counter next to the Usage tab so you can see at a glance
 ![Step templates usage](images/step-templates-usage.png "width=500")
 
-If you have [version-controlled](/docs/projects/version-control/index.md) projects that use step templates, you will be able to see a tab with version-controlled usages in up to twenty recently released projects. You can filter this list to search for usage in a specific branch or use the advanced filters. Step template usage for version-controlled projects is limited to projects (and branches) t ...
+If you have [version-controlled](/docs/projects/version-control/index.md) projects that use step templates, you will be able to see a tab with version-controlled usages in up to twenty recently released projects. **The list of version-controlled usages will only include processes that have been released since converting to version control, and usages will be detected only if a version-controlled  ...
```
```suggestion
If you have [version-controlled](/docs/projects/version-control/index.md) projects that use step templates, you will be able to see a tab with version-controlled usages from up to twenty recent releases. You can filter this list to search for usage in a specific branch or use the advanced filters. 

:::hint
Note: The list of version-controlled usages will only include processes that have been released since converting to version control, and usages will be detected only if a version-controlled process used the step template at the time the release was created.
:::
```

harrisonmeister reviewed (commented):
Made a suggestion on the wording, a hint might be better to highlight the difference.

One other thing; the new image ha a step-template name of `boom`. Any chance we can change it to something else?

LeeMeyer commented:
> Made a suggestion on the wording, a hint might be better to highlight the difference.
> 
> One other thing; the new image ha a step-template name of `boom`. Any chance we can change it to something else?

thank you @harrisonmeister. I moved the clarification into a hint as you said and updated the screenshot with a better step name.
